### PR TITLE
[R4R] implement mimc hash function

### PR DESCRIPTION
**Pull request by forcodedancing**
### Description

Implement native support for MiMC hash function.

### Rationale

MiMC hash function is widely used in blockchain/web3 now, with this feature it will benefit many smart contract developers who are using MiMC in their projects.

### Example

Call the pre-defined contract address for using it.

image

### Changes

Notable changes: 
* add precompiles

### Others

`TODO` Change fork name

### CODE REVIEW

File: core/vm/contracts.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"errors"
 	"math/big"
 
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr/mimc"
```
Comment (yutianwu) [inline core/vm/contracts.go:25]: fix the imports

Comment (forcodedancing) [reply core/vm/contracts.go:25]: updated

**Comment by LBruyne** [comment]
The input parameter of this MiMc function is a string, right?

**Comment by guagualvcha** [comment]
close because seems useless now.

**Pull request closed without merging**